documentation/nearest-neighbor-search.html

```diff
@@ -0,0 +1,436 @@
+ranking function:</p>
+
+<pre>
+rank-profile title-similarity-using-bert {
+  first-phase {
+    expression: closeness("field", title_bert_embedding)
+  }
+}
+</pre> 
+<p>
+Search for 10 nearest neighbors using angular distance in the BERT embedding vector space and we are only interested in products where <em>in_stock</em> is true:
+</p>
+<pre>
+{
+  "yql": "select title,price from sources products where ([{\"targetNumHits\": 10}]nearestNeighbor(title_bert_embedding,query_vector_bert)) and in_stock = true;",
+  "ranking.features.query(query_vector_bert)": [
+    0.7528874147920314,
+    0.712266471788029,
+    0.8813024904379954,
+    ...
+  ],
+  "ranking.profile": "title-similarity-using-bert",
+  "hits": 10
+}
+</pre>
+
+<b>Related Products using hybrid BERT Text Embedding Similarity</b>
+<p> 
+We can also combine and retrieve products in a single query request using disjunction (OR) 
+logical operator to retrieve products which are close in either vector space and we define 
+our ranking function as a weighted linear combination of the closeness scores produced by two nearest neightbor search operators:</p>
+
+<pre>
+rank-profile hybrid-similarity {
+  first-phase {
+    expression: query(bert_weight)*closeness("field", title_bert_embedding) + (1 - query(bert_weight))* closeness("field", image_sift_encoding)
+  }
+}
+</pre>
+<p>Our search request need to pass both the BERT embedding tensor and the SIFT encoding tensor and we use logical disjunction (OR) to combine the
+two nearest neighbor search operators, our search request becomes: </p>
+<pre>
+{
+  "yql": "select title, price from sources products where (([{\"targetNumHits\": 10}]nearestNeighbor(title_bert_embedding,query_vector_bert)) 
+or ([{\"targetNumHits\": 10}]nearestNeighbor(image_sift_encoding,query_vector_sift))) and in_stock = true;",
+  "hits": 10,
+  "ranking.features.query(query_vector_sift)": [
+    0.5158057404746615,
+    0.6441591144976925,
+    0.24075880767944935,
+    ...
+  ],
+  "ranking.features.query(query_vector_bert)": [
+    0.708719978302217,
+    0.4800850502044147,
+    0.03310770782193717,
+    ... 
+  ],
+  "ranking.features.query(bert_weight)": 0.1,
+  "ranking.profile": "hybrid-similarity"
+}
+</pre>
+We ask for 10 nearest neighbors in the title BERT embedding vector space or 10 nearest neighbors in the SIFT encoding space 
+which in total will expose at least 20 documents to the 
+<em>first-phase</em> ranking function. 
+
+<h3 id="hybrid-neural-term-retrieval">Hybrid Neural Term Retrieval</h3>
+<p>In the following example we have a user input query (for example 'white trousers for woman') and we search for both products which matches 
+the query terms and for the nearest neighbors using the neural BERT embedding space.  In this case we need to produce the BERT embedding 
+of the input text query and we use logical disjunction, this allows also retrieving documents which does not use the exact same vocabulary as 
```

Comment:
Consider splitting sentence in two: "...disjunction, this..." -> "...disjunction. This..."